Source/Scene/Expression.js

```diff
@@ -13,14 +17,128 @@ define([
     function Expression(styleEngine, expression) {
         this._styleEngine = styleEngine;
 
-        console.log(jsep(expression));
+        this._ast = jsep(expression);
+        this._runtimeAST = undefined;
+
+        console.log(this._ast);
+        this._runtimeAST = createRuntimeAst(this._ast);
+        console.log(this._runtimeAST);
+    }
+
+    function createRuntimeAst(ast) {
+        var node = ast;
+
+        // evaluate this node if we can
+        if (ast.type === 'Literal') {
+            if (typeof(ast.value) === 'string') {
+                var c = Color.fromCssColorString(ast.value);
+                if (defined(c)) {
+                    node.value = c;
+                }
+            }
+        } else if(ast.type === 'UnaryExpression') {
```

Comment:
These are all similar to the constant folding optimization that I would do later - or never.  I suspect the bulk of this code is going to create problems as we build out the core interpreter for the AST.  I would instead rework this to provide the skeleton needed to evaluate expressions for the runtime AST.  Then later, we can come back and add optimizations like these when creating the tree if needed.  As we discussed, one optimization we will add for sure is to recognize common expressions like `variable operand literal` and then bypass the AST and map it directly to a fast path function.